lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs

```diff
@@ -289,17 +290,55 @@ spec = describe "SHELLEY_TRANSACTIONS" $ do
 
         r <- request @(ApiTransaction n) ctx
             (Link.createTransaction @'Shelley wa) Default payload
-
         verify r
             [ expectSuccess
             , expectResponseCode HTTP.status202
             , expectField (#amount . #getQuantity) $
                 between (feeMin + amt, feeMax + amt)
+            -- all tx inputs have address and amount
+            , expectField #inputs $ \inputs' -> do
+                let addrAmt = (fmap source inputs')
+                addrAmt `shouldSatisfy` (all isJust)
             , expectField (#direction . #getApiT) (`shouldBe` Outgoing)
             , expectField (#status . #getApiT) (`shouldBe` Pending)
             , expectField (#metadata . #getApiTxMetadata) (`shouldBe` Nothing)
             ]
 
+        let txid = getFromResponse #id r
+        let linkSrc = Link.getTransaction @'Shelley wa (ApiTxId txid)
+        let linkDest = Link.getTransaction @'Shelley wb (ApiTxId txid)
+        eventually "transaction is no longer pending and is OK on both wallets" $ do
+            rSrc <- request @(ApiTransaction n) ctx linkSrc Default Empty
+            verify rSrc
+                [ expectSuccess
+                , expectResponseCode HTTP.status200
+                , expectField (#amount . #getQuantity) $
+                    between (feeMin + amt, feeMax + amt)
+                -- all tx inputs have address and amount
+                , expectField #inputs $ \inputs' -> do
+                    let addrAmt = (fmap source inputs')
+                    addrAmt `shouldSatisfy` (all isJust)
+                , expectField (#direction . #getApiT) (`shouldBe` Outgoing)
+                , expectField (#status . #getApiT) (`shouldBe` InLedger)
+                , expectField (#metadata . #getApiTxMetadata) (`shouldBe` Nothing)
+                ]
+
+            rDst <- request @(ApiTransaction n) ctx linkDest Default Empty
+            verify rDst
+                [ expectSuccess
+                , expectResponseCode HTTP.status200
+                , expectField (#amount . #getQuantity) $
+                    (`shouldBe` amt)
+                -- all tx inputs have NO address and amount
+                -- as the tx is incoming
+                , expectField #inputs $ \inputs' -> do
+                    let addrAmt2 = (fmap source inputs')
+                    addrAmt2 `shouldSatisfy` (all isNothing)
```

Comment:
I can confirm this also on `shelley-test-cluster`. So it seems that:
 -  in case of transaction `fixtureWalletSrc` -> `fixtureWalletDest` there are resolved inputs visible on `incoming` transaction on `fixtureWalletDest`.
 - in case of transaction `emptyWallet` -> `fixtureWallet` there are NO resolved inputs visible on `incoming` transaction on `fixtureWallet`

Could this be due to fixtureWallet share the same key or something, so they are able to resolve the inputs in case of transaction between them?

Reply:
Yeah it's a bit fucked up. Since the scenario is using `fixtureWallet`, the UtxO from the wallet comes from one of the fixture transaction made when initializing the cluster. There are many pre-funded fixture wallets with 10 UTxO, but in order not to spend 10 days funding them all, the cluster generate pretty large transaction with ~100 outputs. So a single fixture transaction funds about 10 fixture wallets. 

Now it means that, a particular fixture wallet will know the value of all inputs of its funding fixture transaction and therefore, when picking two fixtureWallet one after the other, it _may_ likely happen that both wallets are able to resolve inputs of any UTxO coming from their original fixture transaction. 

I fixed it by simply using `fixtureWalletWith [...]` which generates completly new and uncorrelated UTxOs.